src/SVGSkin.js

```diff
@@ -60,11 +75,34 @@ class SVGSkin extends Skin {
         super.setRotationCenter(x - viewOffset[0], y - viewOffset[1]);
     }
 
+    /**
+     * Create a MIP for a given scale and pass it a callback for updating
+     * state when switching between scales and MIPs.
+     * @param {number} scale - The relative size of the MIP
+     * @param {function} resetCallback - this is a callback for doing a hard reset
+     * of MIPs and a reset of the rotation center. Only passed in if the MIP scale is 1.
+     * @return {SVGMIP} An object that handles creating and updating SVG textures.
+     */
+    createMIP (scale, resetCallback) {
+        const textureCallback = textureData => {
+            // Check if we have the largest MIP
+            // eslint-disable-next-line no-use-before-define
+            if (!this._scaledMIPs.length || this._scaledMIPs[this._scaledMIPs.length - 1]._scale <= scale) {
+                // Currently silhouette only gets scaled up
+                this._silhouette.update(textureData);
+            }
+
+            if (resetCallback) resetCallback();
```

Comment:
Run resetCallback before checking scaledMIPs.length. As is, this is a race condition. If you change the svgData, the second time, scaledMIPs may have an index higher than 8 (the 1.0 index) set. During that second change the silhouette will not update for the 1.0 scale MIP.

Reply:
Makes sense! Thanks for catching that.